pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -445,6 +454,53 @@ func buildGatewayListenerTLSContext(server *networking.Server, enableSds bool) *
 				},
 			}
 		}
+	} else if server.Tls.Mode == networking.Server_TLSOptions_ISTIO_MUTUAL {
+		// global SDS must be enabled if the sds path is non-default, configure TLS with SDS
+		// TODO would prefer something less implicit here
```

Comment:
If the user configures an `ingress-gateway` with `ISTIO_MUTUAL`, and has workload SDS enabled, this would definitely be an issue. Maybe we should disallow setting `ISTIO_MUTUAL` on `ingress-gateways` through galley validation? Where can we easily find out what kind of gateway we're configuring listeners for?

Reply:
@JimmyCYJ 
Configuring ISTIO_MUTUAL is not always the wrong config for ingress. @vadimeisenbergibm mentioned, e.g. in a multi cluster gateway based setup, workloads in client cluster and send mTLS request to gateway in server cluster using Istio provisioned mTLS. That can be used together with public facing, user provided secrets, on different ports. It seems to be a valid use case to me.

Plus we have a check that workload SDS is enabled before configuring this on gateway.

@vadimeisenbergibm can you confirm?